packages/shared/invokeGuardedCallbackImpl.js

```diff
@@ -183,50 +145,75 @@ if (__DEV__) {
       // Synchronously dispatch our fake event. If the user-provided function
       // errors, it will trigger our global error handler.
       evt.initEvent(evtType, false, false);
-      fakeNode.dispatchEvent(evt);
-
-      if (windowEventDescriptor) {
-        Object.defineProperty(window, 'event', windowEventDescriptor);
-      }
-
-      if (didCall && didError) {
-        if (!didSetError) {
-          // The callback errored, but the error event never fired.
-          // eslint-disable-next-line react-internal/prod-error-codes
-          error = new Error(
-            'An error was thrown inside one of your components, but React ' +
-              "doesn't know what it was. This is likely due to browser " +
-              'flakiness. React does its best to preserve the "Pause on ' +
-              'exceptions" behavior of the DevTools, which requires some ' +
-              "DEV-mode only tricks. It's possible that these don't work in " +
-              'your browser. Try triggering the error in production mode, ' +
-              'or switching to a modern browser. If you suspect that this is ' +
-              'actually an issue with React, please file an issue.',
-          );
-        } else if (isCrossOriginError) {
-          // eslint-disable-next-line react-internal/prod-error-codes
-          error = new Error(
-            "A cross-origin error was thrown. React doesn't have access to " +
-              'the actual error object in development. ' +
-              'See https://reactjs.org/link/crossorigin-error for more information.',
-          );
+      try {
+        fakeNode.dispatchEvent(evt);
+      } catch (dispatchError) {
+        // We don't expect the dispatchEvent to throw directly. If it does something went
+        // terribly wrong. Since we are in the middle of resolving a different error we
+        // suppress this one and let the original error be handled by React.
```

Comment:
@sophiebits you were right, this isn't going to work to still preserve pause behavior. It's a bit baffling b/c the dispatchEvent doesn't throw the inner error but the presence of the catch must signal to devtools that the error was handled (even though it wasn't, it still reports as an uncaught error in the console)

If I extrapolate this though it doesn't make sense that this technique ever worked b/c there is always a catch block somewhere above invokeGuardedCallback so the dispatch is always done in some scope with a catch.

I did a little testing using 18.2 and could not get pause on uncaught exceptions to pause when rendering a component that Errors. Perhaps Chrome changed something and so maybe this used to work and now no longer does b/c browser changes but I'm really not sure

Reply:
My guess is that *we* broke it earlier than 18.2 by introducing a try/catch where we shouldn't have.